### ISSUE TITLE
collapsible_if shouldn't warn on `else` after multiple `else if`

### ISSUE THREAD

**Issue by leoyvens** [rust-lang/rust-clippy#1401]
Consider:
```rust
if cond {
  ...
} else if cond1 {
  bar
} else if cond2 {
  ...
} else if cond3 {
  ...
} else {
  bar
}
```
This will trigger `collapsible_if` because of two equal bodies containing `bar`, but collapsing those ifs would be ugly and error prone, the code is idiomatic as it is.

**Comment by camsteffen** [comment]
Could not [reproduce](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=8e40801616980fa739bd1101a5ca38b6).
